visualization/app/codeCharta/state/store/appSettings/scaling/scaling.actions.ts

```diff
@@ -7,12 +7,12 @@ export enum ScalingActions {
 
 export interface SetScalingAction extends CCAction {
 	type: ScalingActions.SET_SCALING
-	payload: Vector3
+	payload: Partial<Vector3>
```

Comment:
Why did you choose a Partial<Vector3>?

Reply:
Before the heightSettingsPanel.component.ts had to get the state to dispatch a full Vector3 as followed

```ts
applySettingsScaling() {
	const oldScaling = this.storeService.getState().appSettings.scaling
	const newScaling = new Vector3(oldScaling.x, this._viewModel.scalingY, oldScaling.z)

	this.applyDebouncedScaling(newScaling)
}
```

Now it only needs to now about its own y value and doesn't need to now about x and z anymore (`this.store.dispatch(setScaling({ y }))`) and the reducer takes care of keeping the old values.

Reply:
Aaah. Thank you! Now it's clear to me.